### PR TITLE
Remove meaningless lock

### PR DESCRIPTION
**Pull request by garaemon**

### CODE REVIEW

**Comment by garaemon** [comment]
Could someone review this PR? 

Thanks,

File: jsk_interactive_markers/jsk_interactive_marker/src/bounding_box_marker.cpp [hunk 1]
```diff
@@ -57,7 +57,6 @@ void publishClickedBox(jsk_recognition_msgs::Int32Stamped& msg)
 
 void processFeedback(const visualization_msgs::InteractiveMarkerFeedbackConstPtr &feedback)
 {
-  boost::mutex::scoped_lock(mutex);
```
Comment (knorth55) [inline jsk_interactive_markers/jsk_interactive_marker/src/bounding_box_marker.cpp]: do we still need `boost::mutex mutex;` in line 44?

File: jsk_interactive_markers/jsk_interactive_marker/src/polygon_marker.cpp [hunk 2]
```diff
@@ -59,7 +59,6 @@ void publishClickedPolygon(jsk_recognition_msgs::Int32Stamped& msg)
 void processFeedback(
   const visualization_msgs::InteractiveMarkerFeedbackConstPtr &feedback)
 {
-  boost::mutex::scoped_lock(mutex);
```
Comment (knorth55) [inline jsk_interactive_markers/jsk_interactive_marker/src/polygon_marker.cpp]: same as `bounding_box_marker`.
do we still need boost::mutex mutex; in line 45?

File: jsk_interactive_markers/jsk_interactive_marker/src/interactive_point_cloud.cpp [hunk 3]
```diff
@@ -72,7 +72,6 @@ InteractivePointCloud::~InteractivePointCloud(){};
 
 void InteractivePointCloud::configCallback(Config &config, uint32_t level)
 {
-  boost::mutex::scoped_lock(mutex_);
```
Comment (knorth55) [inline jsk_interactive_markers/jsk_interactive_marker/src/interactive_point_cloud.cpp]: do we need `mutex_` in `interactive_point_cloud.h`?

File: jsk_rviz_plugins/src/overlay_diagnostic_display.cpp [hunk 4]
```diff
@@ -112,7 +112,6 @@ namespace jsk_rviz_plugins
   void OverlayDiagnosticDisplay::processMessage(
     const diagnostic_msgs::DiagnosticArray::ConstPtr& msg)
   {
-    boost::mutex::scoped_lock(mutex_);
```
Comment (knorth55) [inline jsk_rviz_plugins/src/overlay_diagnostic_display.cpp]: i think you can remove `mutex_` in `overlay_diagnostic_display.h`.

File: jsk_rviz_plugins/src/overlay_image_display.cpp [hunk 5]
```diff
@@ -149,7 +149,6 @@ namespace jsk_rviz_plugins
   void OverlayImageDisplay::processMessage(
     const sensor_msgs::Image::ConstPtr& msg)
   {
-    boost::mutex::scoped_lock(mutex_);
```
Comment (knorth55) [inline jsk_rviz_plugins/src/overlay_image_display.cpp]: do we need many `boost::mutex::scoped_lock lock(mutex_);` in `OverlayImageDisplay::update~~()`?
I'm not familiar with rviz plugins.

**Comment by knorth55** [comment]
can you tell us purpose of this PR?
is this PR for speed up?
does this PR have backward compatibility for kinetic, melodic and so on?

**Pull request merged**